dev/com.ibm.ws.webcontainer/src/com/ibm/ws/webcontainer/osgi/request/IRequestImpl.java

```diff
@@ -277,6 +279,19 @@ public String getHeader(String headerName)
     return this.request.getHeader(headerName);
   }
 
+  private String getHeader(HttpHeaderKeys key)
+  {
+    return ((HttpRequestExt)this.request).getHeader(key);
```

Comment:
There isn't an exposure of the HttpRequestExt interface to the user.  HttpRequestExt is actually a http transport extension that the web container is calling.  If HttpRequest is still going to be the interface used for interaction with Netty I think it is fair to also implement this interface.  Otherwise if we choose not to do that we can do an instanceof check before making the cast to HttpRequestExt.  Additionally the interface could have a default impl that just calls request.getHeader(key.getName()) so that it can easily be added to the implements interface list for the Netty HttpRequest impl class and not require implementation.

Reply:
@mrsaldana Gary added default implementations in the interfaces, so adding them to the netty impl should be easy.  Could also do instanceof changes in the future if needed as I said previously.